distribution/openrct2.d.ts

```diff
@@ -510,6 +675,598 @@ declare global {
+    }
+
+    interface LandBuyRightsArgs extends GameActionArgs {
+        x1: number;
+        y1: number;
+        x2: number;
+        y2: number;
+        setting: number; // 0: buy land, 1: buy construction rights
+    }
+
+    // x, y specify the centre. Only used for GameActionResult and 3D audio position.
+    // x1, y1, x2, y2 specify a map range
+    interface LandLowerArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        x1: number;
+        y1: number;
+        x2: number;
+        y2: number;
+        selectionType: number; // see MAP_SELECT_TYPE in openrct2/world/Map.h
+    }
+
+    // x, y specify the centre. Only used for GameActionResult and 3D audio position.
+    // x1, y1, x2, y2 specify a map range
+    interface LandRaiseArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        x1: number;
+        y1: number;
+        x2: number;
+        y2: number;
+        selectionType: number; // see MAP_SELECT_TYPE in openrct2/world/Map.h
+    }
+
+    interface LandSetHeightArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        height: number;
+        style: number; // see TILE_ELEMENT_SLOPE in openrct2/world/Surface.h
+    }
+
+    interface LandSetRightsArgs extends GameActionArgs {
+        x1: number;
+        y1: number;
+        x2: number;
+        y2: number;
+        setting: number; // 0: unown land, 1: unown construction rights, 2: set for sale, 3: set construction rights for sale, 4: set ownership
+        ownership: number; // only used if setting = 4 (set ownership). See OWNERSHIP in openrct2/world/Surface.h
+    }
+
+    // x, y specify the centre. Only used for GameActionResult and 3D audio position.
+    // x1, y1, x2, y2 specify a map range
+    interface LandSmoothArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        x1: number;
+        y1: number;
+        x2: number;
+        y2: number;
+        selectionType: number; // see MAP_SELECT_TYPE in openrct2/world/Map.h
+        isLowering: boolean;
+    }
+
+    interface LargeSceneryPlaceArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        z: number;
+        direction: number;
+        object: number;
+        primaryColour: number;
+        secondaryColour: number;
+        tertiaryColour: number;
+    }
+
+    interface LargeSceneryRemoveArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        z: number;
+        direction: number;
+        tileIndex: number;
+    }
+
+    interface LargeScenerySetColourArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        z: number;
+        direction: number;
+        tileIndex: number;
+        primaryColour: number;
+        secondaryColour: number;
+        tertiaryColour: number;
+    }
+
+    interface LoadOrQuitArgs extends GameActionArgs {
+        mode: number; // 0: open save prompt, 1: close save prompt, 2: quit without save prompt
```

Comment:
Internally we never use 2. Its not even in the enum. Its a bit dangerous it will do different things to different users. I suggest not mentioning it here.

Reply:
Makes sense.

In that case, referencing the discussion above, I would be even more okay with this action getting removed from the API. Still, if there is no major reason against it, I would keep the action.

Reply:
if you think it has use to you as a plugin author keep it.